core/dbt/contracts/graph/unparsed.py

```diff
@@ -93,6 +93,8 @@ class HasDocs(AdditionalPropertiesMixin, ExtensibleDbtClassMixin, Replaceable):
     description: str = ""
     meta: Dict[str, Any] = field(default_factory=dict)
     data_type: Optional[str] = None
+    constraint: Optional[str] = None
```

Comment:
Could there ever be multiple constraints? Should this be a list (or [either a string or a list](https://github.com/dbt-labs/dbt-core/blob/ff2f1f42c394d27ff8d49ca7e285fcf2b76faa05/core/dbt/contracts/graph/model_config.py#L122-L126))?

Reply:
Ooooo great question. For developer UX, I'd bias towards a list even though it may be more "efficient" to allow arbitrary strings, developer mental load better reasons about multiple constraints as a list.

**I'll make the change for the `constraints` config to be an optional list**

**String Format:** 
Harder to reason about separate constraints
```
# models/my_custom_model_config.yml
models:
  - name: my_model
    columns:
      - name: email
        data_type: string
        constraints: not null unique 
```

**List Format:** 
Easier to reason about separate constraints
```
# models/my_custom_model_config.yml
models:
  - name: my_model
    columns:
      - name: email
        data_type: string
        constraints: ['not null','unique']
```